### PR TITLE
Arccosh: Numpy Mathematical function 1st commit

### PR DESCRIPTION
**Pull request by deletella01**

### CODE REVIEW

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

Can you fix the lint errors caused by this PR,

Thanks.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_hyperbolic_functions.py [hunk 1]
```diff
@@ -21,14 +21,14 @@
...
 def test_numpy_sinh(
-    dtype_and_x,
-    dtype,
-    where,
-    as_variable,
-    with_out,
-    num_positional_args,
-    native_array,
-    fw,
+        dtype_and_x,
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_hyperbolic_functions.py:24]: shouldn't make changes to other functions in this PR

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

There still seems to be some lint errors in this file,
let em know if you need any help fixing these,

Thanks

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_hyperbolic_functions.py (hunk 1 above)
Comment (deletella01) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_hyperbolic_functions.py:24]: There are no lint errors with the function I worked on, other lint errors are linked to functions that were not written by me. Please I would need further correction and clarification if I am allowed to tamper with other functions. Thanks

File: ivy/functional/frontends/numpy/mathematical_functions/hyperbolic_functions.py [hunk 2]
```diff
@@ -24,3 +24,24 @@ def sinh(
 
 
 sinh.unsupported_dtypes = {"torch": ("float16",)}
+
```
Comment (sherry30) [inline ivy/functional/frontends/numpy/mathematical_functions/hyperbolic_functions.py:27]: add another blank line here

File: ivy/functional/frontends/numpy/mathematical_functions/hyperbolic_functions.py [hunk 3]
```diff
@@ -24,3 +24,24 @@ def sinh(
...
+    subok=True,
+):
+    if dtype:
+        x = ivy.astype(ivy.array(x), ivy.as_ivy_dtype(dtype))
+    ret = ivy.acosh(x, out=out)
+    if ivy.is_array(where):
+        ret = ivy.where(where, ret, ivy.default(out, ivy.zeros_like(ret)), out=out)
+    return ret
+
+arccosh.unsupported_dtypes = {"torch": ("float16",)}
```
Comment (sherry30) [inline ivy/functional/frontends/numpy/mathematical_functions/hyperbolic_functions.py:47]: add a new line after this

File: ivy/functional/frontends/numpy/mathematical_functions/hyperbolic_functions.py [hunk 4]
```diff
@@ -24,3 +24,24 @@ def sinh(
...
+    dtype=None,
+    subok=True,
+):
+    if dtype:
+        x = ivy.astype(ivy.array(x), ivy.as_ivy_dtype(dtype))
+    ret = ivy.acosh(x, out=out)
+    if ivy.is_array(where):
+        ret = ivy.where(where, ret, ivy.default(out, ivy.zeros_like(ret)), out=out)
+    return ret
+
```
Comment (sherry30) [inline ivy/functional/frontends/numpy/mathematical_functions/hyperbolic_functions.py:46]: add another line after this

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

hi I've pointed out the lint errors in your code, you're not supposed to make changes to any other function. You can ignore their lint errors.

Thanks.

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

Can you remove changes to the `sinh` and `test_numpy_sinh` function

Thanks.

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

There are still changes being detected in `test_numpy_sinh`

**Comment by deletella01** [comment]
Mistakenly Closed PR

File: ivy/functional/frontends/numpy/mathematical_functions/hyperbolic_functions.py [hunk 5]
```diff
@@ -5,15 +5,15 @@
...
-    x,
-    /,
-    out=None,
-    *,
-    where=True,
-    casting="same_kind",
-    order="k",
-    dtype=None,
-    subok=True,
+        x,
```
Comment (sherry30) [inline ivy/functional/frontends/numpy/mathematical_functions/hyperbolic_functions.py:8]: remove changes from this function

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

I think you accidentally removed `tanh` function in both test file and ivy/frontend file, can you add it back, your PR is detecting it.

Thanks

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_hyperbolic_functions.py [hunk 6]
```diff
@@ -163,3 +163,54 @@ def test_numpy_arcsinh(
...
+    )
+    np_frontend_helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="numpy",
+        fn_name="arccosh",
```
Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_hyperbolic_functions.py:207]: change this argument to `fn_tree`

Comment (sherry30) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_hyperbolic_functions.py:207]: change this argument to `fn_tree`

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

I pointed out a change on the specific line,
Even after that change, the test seem to fail, can you look into it.
Let me know if you need help with it

Thanks

**Pull request merged**